src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -170,53 +176,54 @@ internal class TypeAccessibilityCodeActionProvider : CSharpCodeActionProvider
 
         private static bool TryProcessCodeActionVS(
             RazorCodeActionContext context,
-            CodeAction codeAction,
+            RazorCodeAction codeAction,
             Diagnostic diagnostic,
-            string associatedValue,
-            out ICollection<CodeAction> typeAccessibilityCodeActions)
+            out ICollection<RazorCodeAction> typeAccessibilityCodeActions)
         {
-            CodeAction processedCodeAction = null;
+            typeAccessibilityCodeActions = new List<RazorCodeAction>(1);
 
-            // When there's only one FQN suggestion, code action title is of the form:
-            // `System.Net.Dns`
-            if (!codeAction.Title.Any(c => char.IsWhiteSpace(c)) &&
-                codeAction.Title.EndsWith(associatedValue, StringComparison.OrdinalIgnoreCase))
-            {
-                var fqn = codeAction.Title;
-                processedCodeAction = CreateFQNCodeAction(context, diagnostic, codeAction, fqn);
-            }
-            // When there are multiple FQN suggestions, the code action title is of the form:
-            // `Fully qualify 'Dns'`
-            else if (codeAction.Title.Equals($"Fully qualify '{associatedValue}'", StringComparison.OrdinalIgnoreCase))
+            if (codeAction.Name.Equals(RazorPredefinedCodeFixProviderNames.FullyQualify, StringComparison.Ordinal))
             {
-                // Not currently supported as we need O# CodeAction to support the CodeAction.Children field.
-                // processedCodeAction = codeAction.WrapResolvableCSharpCodeAction(context, LanguageServerConstants.CodeActions.FullyQualifyType);
-
-                typeAccessibilityCodeActions = Array.Empty<CodeAction>();
-                return false;
+                if (codeAction.Children?.Length == 0)
+                {
+                    // When there's only one FQN suggestion, code action title is of the form:
+                    // `System.Net.Dns`
+                    var fqn = codeAction.Title;
+                    typeAccessibilityCodeActions.Add(CreateFQNCodeAction(context, diagnostic, codeAction, fqn));
+                }
+                else
+                {
+                    // When there are multiple FQN suggestions, the code action title is of the form:
+                    // `Fully qualify 'Dns'`
+                    foreach (var childCodeAction in codeAction.Children)
+                    {
+                        var fqn = childCodeAction.Title;
+                        typeAccessibilityCodeActions.Add(CreateFQNCodeAction(context, diagnostic, childCodeAction, fqn));
+                    }
+                }
```

Comment:
This code needs to be validated still. Blocked on https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1296707

Reply:
Should we loop Etienne in?

Reply:
Repro'd and from my understanding we aren't actually getting any diagnostics from platform.

Discussing with Miguel on the linked thread as I believe this may be a regression from the recent changes.

Reply:
Fixed via: https://github.com/dotnet/aspnetcore-tooling/pull/3361